content/docs/archive-not-matching-git.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Generated code in upstream archives"
+draft: false
+disableToc: false
+weight: 10
+---
+
+# When the release archive does not match the upstream git tag checkout
+
+This document covers a scenario an upstream project has a script to create
+release archives and some of the code in the archive is generated. This means
+that a downstream distribution mat then have downstream patches of the
+generated code which "can't be placed in the upstream repo".
+
+
+## The solution
+
+The ideal solution is to go for [source-git]({{< ref "source-git" >}}).
+
+When all your downstream patches apply cleanly on top of the upstream git repo
+for the particular git tag, then all is good and you don't need anything
+special. You can follow [the guide for the init command]({{< ref "init" >}}) to
+create such a source-git repo.
+
+The real problem rises when you need to patch code which is **not** present in
```

Comment:
I think the real problem (speaking in the context of Ruby) is the opposit case, i.e. there is upstream change, which I'd like to apply downstream, but it would need to went through the preproces script to be applied properly to the tarball. The issue is that I don't have the tools available, e.g. I'd need to have Ruby to be able to successfully execute the preprocess script.